File: python/cops.py
HOUSES = 100
T = int(input())
safe_count = []

def getSafeCount(N, x, y, cops):
    danger_houses = []
    for house in cops:
        l_safe = 1 if house - (x * y) < 1 else house - (x * y)
        r_safe = HOUSES if house + (x * y) > HOUSES else house + (x * y)
        danger_houses = danger_houses + list(range(l_safe, r_safe + 1))
    return HOUSES - len(set(danger_houses))

for _ in range(T):
    N, x, y = map(int, input().split())
    cops = list(map(int, input().split()))
    safe_count.append(getSafeCount(N, x, y, cops))

for count in safe_count:
    print(count)

File: java/Cops.java
import java.util.Scanner;
class Cops {
	int numOfCops;
	int dist;
	public final int HOUSES = 100;
	public final boolean SAFE = true;
	public final boolean UNSAFE = false;
	Cops() {
	}
	public void setValues(int numOfCops, int speed, int time) {
		this.numOfCops = numOfCops;
		this.dist = speed * time;
	}
	public int getSafeCount(int[] cops) {
		boolean[] house = new boolean[HOUSES];
		int safeCount = HOUSES;
		for(int i = 0; i < HOUSES; i++) {
			house[i] = SAFE;
		}
		for(int i = 0; i < numOfCops; i++) {
			int lSafe = cops[i] - dist - 1;
			int rSafe = cops[i] + dist - 1;
			if(lSafe < 0)	lSafe = 0;
			if(rSafe >= HOUSES)	rSafe = HOUSES - 1;
			for(int j = lSafe; j <= rSafe; j++) {
				if(house[j])	safeCount--;
				house[j] = UNSAFE;
			}
		}
		return safeCount;
	}
	public static void main(String args[]) {
		Scanner s = new Scanner(System.in);
		int T = s.nextInt();
		Cops cops = new Cops();
		int[] safeCount = new int[T];
		for(int i = 0; i < T; i++) {
			int numOfCops = s.nextInt();
			int speed = s.nextInt();
			int time = s.nextInt();
			cops.setValues(numOfCops, speed, time);
			int[] copsList = new int[numOfCops];
			for(int j = 0; j < numOfCops; j++) {
				copsList[j] = s.nextInt();
			}
			safeCount[i] = cops.getSafeCount(copsList);
		}
		for(int i = 0; i < T; i++) {
			System.out.println(safeCount[i] + " ");
		}
	}
}
